pyhf/__init__.py

```diff
@@ -2,10 +2,29 @@
 import pyhf.optimize as optimize
 import pyhf.tensor as tensor
 
-
 log = logging.getLogger(__name__)
 tensorlib = tensor.numpy_backend()
+default_backend = tensorlib
 optimizer = optimize.scipy_optimizer()
+default_optimizer = optimizer
+
+def get_backend():
```

Comment:
hi @kratsg,

why is this needed? Shouldn't `pyhf.tensorlib` and `pyhf.optimizer` always point to the current backend?

Reply:
Yeah, it turns out that the class definitions in a separate subpackage or somehow caching or seeing the old value of `tensorlib`, even if it changes... If I use `tensorlib` outright and not `get_backend()`, the tests do not pass.

I think `pyhf.tensorlib` will always be right, but we're using relative imports for this so if we're keeping the import style consistent... then the cache'ing happens with relative imports.